03-amazon_s3.Rmd

```diff
@@ -0,0 +1,244 @@
+To add a new user to the data access group:
+
+1.  Type the user's GitHub username into the input field labelled __Grant access to this data to other users__.
+2.  Select the user from the drop-down list.
+3.  Select __Grant access__.
+
+To edit the access level of a user:
+
+1.  Select __Edit access level__ next to the name of the user.
+2.  Select the checkbox next to the access level you wish to grant the user.
+3.  Select __Save__.
+
+To remove a user from the data access group:
+
+1.  Select __Edit access level__ next to the name of the user.
+2.  Select __Revoke access__.
+
+## Upload files to Amazon S3
+
+You can upload files to Amazon S3 from your local computer or from RStudio or JupyterLab.
+
+When uploading files to Amazon S3, you should ensure that you follow all necessary [information governance]() procedures. In particular, you must complete a data movement form when moving any data onto the Analytical Platform.
+
+### Amazon S3 Console
+
+You can use the Amazon S3 Console to upload files from your local computer (e.g., personal or shared storage on DOM1 or Quantum) only.
+
+To upload files using the Amazon S3 Console:
+
+1.  Log in to the [AWS Management Console](https://http://aws.services.alpha.mojanalytics.xyz) using your Analytical Platform account.
+2.  Select __Services__ from the menu bar.
+3.  Select __S3__ from the drop down menu.
+4.  Select the folder you want to upload files to.
+5.  Select __Upload__.
+6.  Select __Add files__ or drag and drop the files you want to upload.
+7.  Select __Upload__ -- you do not need to complete steps 2, 3 or 4.
+
+You can also directly navigate to a bucket in the AWS S3 Console by selecting __Open on AWS__ in the Analytical Platform control panel.
+
+### RStudio
+
+You can upload files in RStudio on the Analytical Platform to Amazon S3 using the `s3tools` package.
+
+`s3tools` should be preinstalled for all users of the Analytical Platform. If you find that `s3tools` is not installed, you can install it by running the following code:
+
+```{r install-s3-tools, eval=FALSE}
+install.packages('remotes')
+library(remotes)
+remotes::install_github('moj-analytical-services/s3tools')
+```
+
+`s3tools` contains three functions for uploading files to Amazon S3:
+
+*   `write_file_to_s3`
+*   `write_df_to_csv_in_s3`
+*   `write_df_to_table_in_s3`
+
+You can find out more about how to use these functions on [GitHub](https://github.com/moj-analytical-services/s3tools) or by using the help operator in RStudio (e.g., `?s3tools::write_file_to_s3`).
+
+### JupyterLab
+
+You can upload files in JupyterLab on the Analytical Platform to Amazon S3 using the `boto3` package.
+
+You can install `boto3` by running the following code in a terminal:
+
+```{python install-boto3-1, eval=FALSE}
+pip install boto3
+```
+
+To upload a file to Amazon S3, use the following code:
+
+```{python python-upload-s3, eval=FALSE}
+import boto3
+
+s3 = boto3.resource('s3')
+s3.object('bucket_name', 'key').put(Body=object)
+```
```

Comment:
If you get `ImportError` then try restarting your kernel, so that it recognizes your recent install of boto3.

Reply:
Added in 26514c9.